graphics/src/widget/canvas/frame.rs

```diff
@@ -107,25 +109,24 @@ impl Frame {
                 &options,
                 &mut buffers,
             )
-        };
+        }.expect("Tessellate path.");
 
-        result.expect("Tessellate path");
+        self.buffers.push((buf, style.into()))
     }
```

Comment:
The Gradient start / stop isn't currently affected by the current transform. So if we are not on the identity transform, the gradient direction vector will be inaccurate compared to this path. Instead of having a `From<(Stroke|Fill)Style> for Shader`, can there instead be a `build` type method that accepts `Transform`? We can then expose a `Transform::transform_point` helper and apply it against the gradients start / stop before storing it as a shader.

Reply:
or something like `Gradient::from(style).transform(self.transforms.current)`